v2/internal/testcommon/resource_namer.go

```diff
@@ -18,36 +19,65 @@ type ResourceNameConfig struct {
 	prefix      string
 	randomChars int
 	separator   string
+	mode        ResourceNamerMode
 }
 
+type ResourceNamerMode string
```

Comment:
This seems familiar - was it in the other PR as well?

Reply:
Yes, this PR and the other PR are "the same" except this one isn't from my fork it's from a branch of the Azure repo. Once this is merged I'll open a new PR from my fork to make sure the fork workflow works.